### PR TITLE
Add path to active developer directory to commandline options

### PR DESCRIPTION
**Pull request by akoslowski**
Currently the path to `libIndexStore` is hard-coded to `/Applications/Xcode.app/...`. Working with different versions of Xcode requires selecting the Xcode to use. This PR adds an option to the command line interface.

```bash
pecker [...] --active-developer-directory "/Applications/Xcode-13.3.app/Contents/Developer" [...]

pecker [...] --active-developer-directory "$(xcode-select -p)" [...]
```

```bash
▶ .build/debug/pecker --help
OVERVIEW: Detect unused Swift and Objective-C code

USAGE: Pecker [--path <path>] [--index-store-path <index-store-path>] [--config <config>] [--active-developer-directory <active-developer-directory>]

OPTIONS:
  --path <path>           Path of the project to detect
  -i, --index-store-path <index-store-path>
                          Specify project index path [default:
                          ~/Library/Developer/Xcode/DerivedData]
  --config <config>       The path of the configuration file
  --active-developer-directory <active-developer-directory>
                          The path of the active developer directory; Using 'xcode-select
                          -p' is recommended (default:
                          /Applications/Xcode.app/Contents/Developer)
  --version               Show the version.
  -h, --help              Show help information.
```

### CODE REVIEW

File: Sources/PeckerKit/Configuration.swift [hunk 1]
```diff
@@ -77,6 +81,8 @@ public struct Configuration {
         let excluded = (yamlConfiguration?.excluded ?? []).map {
             return AbsolutePath($0, relativeTo: projectPath)
         }.filter { localFileSystem.exists($0) }
+
+        let xcodeAppPath = createXcodeAppPath(path: yamlConfiguration?.xcodeAppPath, fileSystem: localFileSystem) ?? AbsolutePath("/Applications/Xcode.app")
```
Comment (sdidla) [inline Sources/PeckerKit/Configuration.swift]: `AbsolutePath("/Applications/Xcode.app")`:  what do you think of making this an even smarter default by running `xcode-select -p` and getting the path :)

Comment (akoslowski) [reply Sources/PeckerKit/Configuration.swift]: Good point. `PeckerKit` is probably not the right place to add a call to `xcode-select`. Maybe we can add an option to the commandline tool and then you can call the tool with `pecker --xcode-app-path $(xcode-select -p)`. What do you think?

Comment (akoslowski) [reply Sources/PeckerKit/Configuration.swift]: The output of `xcode-select -p` includes `.../Contents/Developer/`. Should we be smart and check the given path and append `/Contents/Developer/` if it is missing? 

While writing I already don't like the idea. 😄 I think requiring the path that `xcode-select -p` gives you is the better idea.

File: Sources/PeckerKit/Configuration.swift [hunk 2]
```diff
@@ -87,10 +97,28 @@ public struct Configuration {
...
     }
 }
 
+private func createXcodeAppPath(pathFromOptions: String?, pathFromConfig: String?, fileSystem: FileSystem) -> AbsolutePath {
+    if let absolutePath = absolutePathIfExists(path: pathFromOptions, fileSystem: localFileSystem) {
+        return absolutePath
+    } else if let absolutePath = absolutePathIfExists(path: pathFromConfig, fileSystem: localFileSystem) {
+        return absolutePath
+    } else {
+        return AbsolutePath("/Applications/Xcode.app")
```
Comment (akoslowski) [inline Sources/PeckerKit/Configuration.swift]: Now that we have a command line options, we could move the default value into `Sources/Pecker/PeckerCommand.swift`

**Review by sdidla — APPROVED** [review]
Nice!

**Pull request merged**